plugin/src/main/kotlin/me/tylerbwong/gradle/metalava/plugin/MetalavaPlugin.kt

```diff
@@ -6,25 +6,33 @@ import me.tylerbwong.gradle.metalava.task.MetalavaCheckCompatibility
 import me.tylerbwong.gradle.metalava.task.MetalavaSignature
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import java.util.Locale
 
 class MetalavaPlugin : Plugin<Project> {
     override fun apply(target: Project) {
         with(target) {
             val extension = extensions.create("metalava", MetalavaExtension::class.java)
             afterEvaluate {
                 val currentModule = module(extension)
-                MetalavaSignature.registerMetalavaSignatureTask(
-                    project = this,
-                    name = "metalavaGenerateSignature",
-                    description = "Generates a Metalava signature descriptor file.",
-                    extension = extension,
-                    module = currentModule
-                )
-                MetalavaCheckCompatibility.registerMetalavaCheckCompatibilityTask(
-                    project = this,
-                    extension = extension,
-                    module = currentModule
-                )
+                val taskVariants = currentModule.taskVariants.ifEmpty { listOf("") }
+                taskVariants.forEach {
+                    val variantName = it.capitalize(Locale.getDefault())
```

Comment:
Do we need to do any extra handling for Metalava to be able to look at a specific variant or is it handled automatically somehow?

Reply:
I don't think there is. The change allows the association of the variant tasks (`metalavaXDebug` etc) to be configured according to how the including project is set up, really. From my point of view, that's to allow that project to provide the metalava plugin with different source dirs based on the variant. metalava itself (I don't think?) needs to do anything different - it's just parsing the source that it's told to. Even if there isn't a need to differentiate the source input dirs, based on the variant, the plugin now (in this change) would align with what variants Android would expect (ie your debug and release builds).